plugins/platform-integrity/fu-plugin-platform-integrity.c

```diff
@@ -10,17 +10,47 @@
 #include "fu-plugin-vfuncs.h"
 #include "fu-hash.h"
 
-#define FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR	"/sys/kernel/security/spi"
+struct FuPluginData {
+	gchar			*sysfs_path;
+};
 
 void
 fu_plugin_init (FuPlugin *plugin)
 {
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
 	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_add_udev_subsystem (plugin, "platform-integrity");
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+	g_free (priv->sysfs_path);
+}
+
+gboolean
+fu_plugin_udev_device_added (FuPlugin *plugin, FuUdevDevice *device, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+
+	/* interesting device? */
+	if (g_strcmp0 (fu_udev_device_get_subsystem (device), "platform-integrity") != 0)
+		return TRUE;
+
+	/* we only care about the first instance */
+	if (priv->sysfs_path != NULL)
+		return TRUE;
```

Comment:
Aren't these supposed to return FALSE and have an error?

Reply:
I'm not sure. In this case this is a sysfs class that will only realistically have one thing implementing it. Maybe an error is a good idea as then we can see if this assumption is ever voided.

Reply:
There are certainly multiple things that can provide a view of platform integrity, but how much is expressed in open source, I don't know.  I think it's better to switch to FALSE and show an error in case that ever happens.

Reply:
> but how much is expressed in open source, I don't know

Reply:
`NOT_SUPPORTED` seems fine to me, this isn't useful by default to anyone other than a debugging capacity.

Reply:
Okay, fixed.